WearVerifyRemoteApp/Wearable/src/main/java/com/example/android/wearable/wear/wearverifyremoteapp/MainWearActivity.kt

```diff
@@ -0,0 +1,206 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.android.wearable.wear.wearverifyremoteapp
+
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.util.Log
+import androidx.core.view.doOnPreDraw
+import androidx.core.view.isInvisible
+import androidx.core.view.isVisible
+import androidx.core.view.setPadding
+import androidx.fragment.app.FragmentActivity
+import androidx.lifecycle.lifecycleScope
+import androidx.wear.phone.interactions.PhoneTypeHelper
+import androidx.wear.remote.interactions.RemoteIntentHelper
+import androidx.wear.widget.ConfirmationOverlay
+import com.example.android.wearable.wear.wearverifyremoteapp.databinding.ActivityMainBinding
+import com.google.android.gms.wearable.CapabilityClient
+import com.google.android.gms.wearable.CapabilityInfo
+import com.google.android.gms.wearable.Node
+import com.google.android.gms.wearable.NodeClient
+import com.google.android.gms.wearable.Wearable
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.guava.await
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.tasks.await
+import kotlinx.coroutines.withContext
+import kotlin.math.roundToInt
+import kotlin.math.sqrt
+
+/**
+ * Checks if the phone app is installed on remote device. If it is not, allows user to open app
+ * listing on the phone's Play or App Store.
+ */
+class MainWearActivity : FragmentActivity(), CapabilityClient.OnCapabilityChangedListener {
+
+    private lateinit var binding: ActivityMainBinding
+
+    private lateinit var capabilityClient: CapabilityClient
+    private lateinit var nodeClient: NodeClient
+    private lateinit var remoteIntentHelper: RemoteIntentHelper
+
+    private var androidPhoneNodeWithApp: Node? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        Log.d(TAG, "onCreate()")
+        super.onCreate(savedInstanceState)
+
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        capabilityClient = Wearable.getCapabilityClient(this)
+        nodeClient = Wearable.getNodeClient(this)
+        remoteIntentHelper = RemoteIntentHelper(this)
+
+        binding.informationTextView.text = getString(R.string.message_checking)
+        binding.remoteOpenButton.setOnClickListener { 
+            openAppInStoreOnPhone()
+        }
+
+        if (resources.configuration.isScreenRound) {
+            binding.scrollingContentContainer.doOnPreDraw {
+                // Calculate the padding necessary to make the scrolling content fit in a square inscribed on a round
+                // screen.
+                it.setPadding((it.width / 2.0 * (1.0 - 1.0 / sqrt(2.0))).roundToInt())
+            }
+        }
+    }
+
+    override fun onPause() {
+        Log.d(TAG, "onPause()")
```

Comment:
I would remove Log messages that don't print something related to the sample (unless the sample is demonstrating a certain timing or execution, but I don't get that impression here).

Reply:
That makes sense to me, I left them in since they were there previously, took out the lifecycle ones with 5e65373.